src/lexer/lex.c

```diff
@@ -1,10 +1,27 @@
 #include "lex.h"
 
+#include <ctype.h>
 #include <string.h> // memcpy
 #include <assert.h> // assert
 
 #include <util/out.h> // error reporting
 
+// Is a character in the given string?
+int in_string(char c, char s[]) {
+    for (char* d = s; *d; ++d) {
+        if (*d == c)
+            return 1;
+    }
+    return 0;
+}
+
+// We will need to add more of these later, for sure
+char single_char_tokens[] = "(){}[];";
```

Comment:
Could we use a 64b bit field for this?

Reply:
What would the code look like for that? I’m not convinced it would be faster and besides, we will have many many more single character strings soon so it won’t fit them all